webapi/src/main/scala/org/knora/webapi/messages/v1/responder/groupmessages/GroupMessagesV1.scala

```diff
@@ -0,0 +1,288 @@
+// Components of messages
+
+/**
+  * The information describing a group.
+  *
+  * @param id the IRI if the group.
+  * @param name the name of the group.
+  * @param description the description of the group.
+  * @param belongsToProject the project this group belongs to.
+  * @param isActiveGroup the group's status.
+  * @param hasSelfJoinEnabled the group's self-join status.
+  * @param hasPermissions the permissions attached to the group.
+  *
+  */
+case class GroupInfoV1(id: IRI,
+                       name: String,
+                       description: Option[String] = None,
+                       belongsToProject: IRI,
+                       isActiveGroup: Boolean = false,
+                       hasSelfJoinEnabled: Boolean = false,
+                       hasPermissions: Seq[GroupPermissionV1] = Vector.empty[GroupPermissionV1]) {
+
+    def ofType(groupInfoType: GroupInfoType): GroupInfoV1 = {
+
+        groupInfoType match {
+            case GroupInfoType.FULL => {
+                GroupInfoV1(
+                    id = id,
+                    name = name,
+                    description = description,
+                    belongsToProject = belongsToProject,
+                    isActiveGroup = isActiveGroup,
+                    hasSelfJoinEnabled = hasSelfJoinEnabled,
+                    hasPermissions = hasPermissions
+                )
+            }
+            case GroupInfoType.SAFE => {
+                GroupInfoV1(
+                    id = id,
+                    name = name,
+                    description = description,
+                    belongsToProject = belongsToProject,
+                    isActiveGroup = isActiveGroup,
+                    hasSelfJoinEnabled = hasSelfJoinEnabled,
+                    hasPermissions = Vector.empty[GroupPermissionV1]
+                )
+            }
+            case _ => throw BadRequestException(s"The requested groupInfoType: $groupInfoType is invalid.")
+        }
+    }
+}
+
+
+/**
+  * Represents basic information about the group which need to be supplied during group creation.
+  *
+  * @param name the name of the group.
+  * @param description the optional description of the group
+  * @param belongsToProject the project this group belongs to.
+  * @param isActiveGroup the group's status.
+  * @param hasSelfJoinEnabled the group's self-join status.
+  */
+case class NewGroupInfoV1(name: String,
+                          description: Option[String] = None,
+                          belongsToProject: IRI,
+                          isActiveGroup: Boolean = true,
+                          hasSelfJoinEnabled: Boolean = false)
+
+/**
+  * Represents a group permission.
+  * @param name the name of the permission.
+  * @param value the value of the permission.
+  */
+case class GroupPermissionV1(name: IRI, value: Either[Boolean, List[IRI]])
+
+object GroupInfoType extends Enumeration {
```

Comment:
The permission information should only be returned if the user has the necessary permission. So if someone doesn't have the permission to see everything, we would only return what he is allowed to see, hence ``SAFE ``, ``FULL``. The naming is probably not the best. I should've named this something like ``FULL``, ``RESTRICTED``.